\section*{Abstract}

The expression of genes in the human genome is not only based on the DNA sequence; it relies on epigenetic modifications like DNA-methylation. Hereby, the gene expression is inactivated by the binding of methyl-groups to the cytosine of a \ac{CpG} dinucleotide at the promoter region of the concerned gene. The binding is performed by specific enzymes - the \acp{DNMT}.\\

In this thesis, we model the methylation of DNA using an \ac{MCMC} algorithm and estimate the corresponding parameters with \ac{MLE}. Alternatively, parameter estimation is performed with an implementation of the \ac{ABC} method. Moreover, different distance functions are provided to compare distributions of methylation patterns.\\
We find differences in the function of the different \acp{DNMT} that are consistent with current biological findings. Using this model, some parameters are difficult to identify because they seem to be conditionally dependant.

\newpage
\mbox{}
\newpage